File: scripts/binary_search.py
def binary_search(arr: list[int], num: int) -> int | None:
    low = 0
    high = len(arr) - 1

    while low <= high:
        mid = (low + high) // 2
        guess = arr[mid]
        if guess == num:
            return guess
        elif guess > num:
            high = mid - 1
        elif guess < num:
            low = mid + 1
    return None

File: javascript/binarySearch.js
function binarySearch(arr, num) {
    let low = 0
    let high = arr.length - 1

    while (low <= high) {
        let mid = Math.floor((high + low) / 2)
        let guess = arr[mid]

        if (guess === num) {
            return mid
        } else if (guess < num) {
            low = mid + 1
        } else if (guess > num) {
            high =  mid - 1
        }
    }

    return -1
}

const arr = [1, 2, 3,4, 5]
console.log(binarySearch(arr, 5))